sqlglot/optimizer/simplify.py

```diff
@@ -881,6 +885,23 @@ def simplify_datetrunc_predicate(expression: exp.Expression) -> exp.Expression:
     return expression
 
 
+def sort_comparison(expression: exp.Expression) -> exp.Expression:
+    if expression.__class__ in COMPLEMENT_COMPARISONS:
+        l, r = expression.this, expression.expression
+        l_column = isinstance(l, exp.Column)
+        r_column = isinstance(r, exp.Column)
+        l_literal = isinstance(l, (exp.Literal, exp.Boolean))
+        r_literal = isinstance(r, (exp.Literal, exp.Boolean))
+
+        if (l_column and not r_column) or (r_literal and not l_literal):
+            return expression
+        if (r_column and not l_column) or (l_literal and not r_literal) or (gen(l) > gen(r)):
+            return INVERSE_COMPARISONS.get(expression.__class__, expression.__class__)(
```

Comment:
This looks like a typo? Did you mean `COMPLEMENT_COMPARISONS` or am I misreading this?

Reply:
it's not a typo, you don't want to invert = when sorting 

a = b == b = a

Reply:
Ah I didn't see that you do `get(expression.__class__, expression.__class__)`. Makes sense, thanks for clarifying.

Perhaps we could also clean up some of these `__class__` attributes by doing `type(expression)`?